### DISCUSSION TITLE
broken (?) typing on JWTStrategy

### DISCUSSION THREAD

**Discussion by RemiCardona** [fastapi-users/fastapi-users#1412] [Q&A]
## Describe the bug

mypy can't seem to figure out the type of the `JWTStrategy` class on its own. The full example in the tutorial doesn't raise any typing error as the class is used within a function with a typed return value.

When assigned to a standalone variable, typing issues occur. Below is an example with a comparison with `CookieTransport` which doesn't raise any typing errors.

## To Reproduce

Steps to reproduce the behavior, in `blah.py` below:

```python
from fastapi_users.authentication import CookieTransport, JWTStrategy

cookie_transport = CookieTransport(cookie_max_age=3600)
jwt_strategy = JWTStrategy(secret="SECRET", lifetime_seconds=3600)
```

```console
$ mypy blah.py 
blah.py:4: error: Need type annotation for "jwt_strategy"  [var-annotated]
Found 1 error in 1 file (checked 1 source file)
```

I apparently need to explicitly mention the type to make mypy happy

```python
jwt_strategy: JWTStrategy = JWTStrategy(secret="SECRET", lifetime_seconds=3600)
```

Disclaimer: I'm a total typing noob.

I tried removing the `Generic` type on `JWTStrategy` but that doesn't seem to help. I have no idea what the correct solution is.

## Expected behavior

No mypy error, as with `CookieTransport`.

## Configuration
 - Python version : 3.11.9
 - FastAPI version : 0.109.2
 - FastAPI Users version : 13.0.0

### FastAPI Users configuration

N/A

## Additional context

N/A

**Comment by mahenzon** [comment]
Hi. `JWTStrategy` is a Generic, so you need to provide types:

```python
strategy = JWTStrategy[User, UserIdType](secret="abc", lifetime_seconds=3600)
```
`User` is a subclass of `SQLAlchemyBaseUserTable`
`UserIdType` is User's id type: UUID or any other type like int / str
